EpsilonAuxBMS/Src/UpdateChargeAllowanceTask.c

```diff
@@ -20,26 +20,100 @@ void updateChargeAllowanceTask(void const* arg)
            and the cell with the lowest voltage. Must check if the voltage of the cell with the highest
            voltage is too high or in range and if the cell with the lowest voltage is too low or in range.
         */
+        if (orionStatus.shutOff)
+        {
+            continue;
+        }
 
         uint8_t voltagesInRange = 1;
         uint8_t allowCharge = 1;
         uint8_t allowDischarge = 1;
+        uint8_t chargeContactorOverride = 0;
+        uint8_t dischargeContactorOverride = 0;
+        uint8_t commonContactorOff = 0;
+        uint8_t shutOff = 0;
 
-        if (DEFAULT_VOLTAGE_UNITS * orionStatus.maxCellVoltage > MAX_CELL_VOLTAGE * 0.8) // Will have the cutoff to be 20% below
+        if (auxStatus.startUpSequenceDone)
+        {
+            if (driversInput.forwardReverse) // In forward or reverse
+            {
+                if (!HAL_GPIO_ReadPin(CHARGE_ENABLE_SENSE_GPIO_Port, CHARGE_ENABLE_SENSE_Pin) ||
+                        !HAL_GPIO_ReadPin(DISCHARGE_ENABLE_SENSE_GPIO_Port, DISCHARGE_ENABLE_SENSE_Pin))
+                {
+                    chargeContactorOverride = 1;
+                    dischargeContactorOverride = 1;
+                    commonContactorOff = 1;
+                    allowCharge = 0;
+                    allowDischarge = 0;
+                    shutOff = 1;
+                    // Turn all contactors and high voltage enable off
+                    HAL_GPIO_WritePin(HV_ENABLE_GPIO_Port, HV_ENABLE_Pin, GPIO_PIN_RESET);
+                    HAL_GPIO_WritePin(COMMON_CONTACTOR_ENABLE_GPIO_Port, COMMON_CONTACTOR_ENABLE_Pin, GPIO_PIN_RESET);
+                    HAL_GPIO_WritePin(CHARGE_CONTACTOR_ENABLE_GPIO_Port, CHARGE_CONTACTOR_ENABLE_Pin, GPIO_PIN_RESET);
+                    HAL_GPIO_WritePin(DISCHARGE_CONTACTOR_ENABLE_GPIO_Port, DISCHARGE_CONTACTOR_ENABLE_Pin, GPIO_PIN_RESET);
```

Comment:
All of this looks a lot like `SetContactorsTask.c:disconnectContactors` any chance for some code reuse?